package.json

```diff
@@ -80,10 +82,11 @@
   },
   "scripts": {
     "install": "composer install",
-    "build": "gulp",
+    "build": "gulp buildDev",
+    "dev": "gulp",
     "translate": "gulp translate",
-    "bundle": "gulp bundleTheme",
-    "test": "gulp testTheme",
-    "generateCert": "gulp generateCert"
+    "bundle": "NODE_ENV=production gulp bundleTheme",
+    "generateCert": "gulp generateCert",
+    "devStringReplace": "gulp sourceStringReplacement"
```

Comment:
I don't like having this task here. It's not of value for the intended workflow, and more importantly can cause problems with that workflow. Since it furthermore adds extra overhead to this PR, I'd prefer to remove it. Sure, someone might wanna approach it this way, but that's not the intention, and it could always be implemented in the respective projects specifically.

_If_ I'm in the minority with saying that it should be removed, I'd at least prefer to only have this task present at the `gulp` level, but not among the other "encouraged" tasks in `package.json`.

Reply:
I'll let @mor10 chime in here since I think he was the one who requested this be possible.

> If I'm in the minority with saying that it should be removed, I'd at least prefer to only have this task present at the gulp level, but not among the other "encouraged" tasks in package.json.

The items in `package.json` aren't just there to be the "encouraged" items. Running gulp with [npm run script]()https://docs.npmjs.com/cli/run-script) allows for the gulp binary installed with npm to be used, rather than requiring the user to install gulp globally.

That being said you can pass extra arguments to items run with npm run script. I added a generic gulp script to npm scripts so that `npm run gulp` can be used. Then, any gulp task can be invoked. For example, `npm run gulp -- images` would run the image task.

Now we can remove `devStringReplace` but keep the gulp task if needed.

Reply:
@felixarntz Unless you object, let's handle the decision to keep or remove this task in a follow-up issue/PR. It would be a small change so I don't think we should block merging this PR.